casatables/src/lib.rs

```diff
@@ -529,6 +529,50 @@ where
     )
 }
 
+// TableDesc
+
+pub struct TableDesc {
+    handle: *mut glue::GlueTableDesc,
+    exc_info: glue::ExcInfo,
+}
+
+impl TableDesc {
+    pub fn new(stype: &str) -> Self {
+        let ctype = glue::StringBridge::from_rust(stype);
+        let exc_info = unsafe { std::mem::zeroed::<glue::ExcInfo>() };
+
+        let handle = unsafe { glue::tabledesc_create(&ctype) };
+
+        // Ok(TableDesc {
+        //     handle,
+        //     exc_info
+        // })
+        TableDesc { handle, exc_info }
+    }
+
+    pub fn add_scalar_column(
+        &mut self,
+        data_type: glue::GlueDataType,
+        col_name: &str,
+    ) -> Result<Self, Error> {
+        let cname = glue::StringBridge::from_rust(col_name);
+        let new_handle = unsafe {
+            glue::tabledesc_add_scalar_column(self.handle, data_type, &cname, &mut self.exc_info)
+        };
+
+        if new_handle.is_null() {
+            return self.exc_info.as_err();
+        }
+
+        // not sure what to do with new_handle after this?
```

Comment:
As far as I can see, the `new_handle` variable is just a redundant copy of the handle, so I think it can be ignored (modulo checking for null). It's just convenient to write object-oriented functions that return the "self" object so that method calls can be chained easily.

Reply:
it felt kind of messy to return `Ok(*self)` so I just changed the return type to `Result<(), Error>`

Reply:
I could also go down the builder pattern route, and have something like this:

```rust
    pub fn add_scalar_column(
        &mut self,
        data_type: glue::GlueDataType,
        col_name: &str,
    ) -> Result<&mut Self, Error> {
        let cname = glue::StringBridge::from_rust(col_name);
        let new_handle = unsafe {
            glue::tabledesc_add_scalar_column(self.handle, data_type, &cname, &mut self.exc_info)
        };

        if new_handle.is_null() {
            return self.exc_info.as_err();
        }

        Ok(self)
    }
```

but then I would have to implement a `finalize()` and move the error handling into it.

```rust
let table_desc = TableDesc::new("TYPE")
    .add_scalar_column(GlueDataType::TpString, "string")
    .add_array_column(GlueDataType::TpUInt, "uint array", 2)
    .finalize();
```

What do you think would be an ergonomic way of doing this?

Another idea could be to add a `.create(type, mode, n_rows)` that creates table from the table description.